test/csuite/wt8057_compact_stress/main.c

```diff
@@ -0,0 +1,381 @@
+  subtest_error_handler, NULL, /* Message handler */
+  NULL,                        /* Progress handler */
+  NULL                         /* Close handler */
+};
+
+static void sig_handler(int) WT_GCC_FUNC_DECL_ATTRIBUTE((noreturn));
+
+/* Forward declarations. */
+static void run_test(const char *home);
+static void populate(WT_SESSION *session, int start, int end);
+static void remove_records(WT_SESSION *session, const char *uri, int start, int end);
+static void verify_tables(WT_SESSION *session);
+static uint64_t get_file_size(WT_SESSION *session, const char *uri);
+
+/*
+ * Signal handler to catch if the child died unexpectedly.
+ */
+static void
+sig_handler(int sig)
+{
+    pid_t pid;
+
+    WT_UNUSED(sig);
+    pid = wait(NULL);
+    /*
+     * The core file will indicate why the child exited. Choose EINVAL here.
+     */
+    testutil_die(EINVAL, "Child process %" PRIu64 " abnormally exited", (uint64_t)pid);
+}
+
+/* Methods implementation. */
+int
+main(int argc, char *argv[])
+{
+    WT_CONNECTION *conn;
+    WT_SESSION *session;
+
+    struct sigaction sa;
+    struct stat sb;
+    TEST_OPTS *opts, _opts;
+    char home[1024];
+    int status;
+
+    pid_t pid;
+
+    opts = &_opts;
+    memset(opts, 0, sizeof(*opts));
+    testutil_check(testutil_parse_opts(argc, argv, opts));
+
+    testutil_work_dir_from_path(home, sizeof(home), working_dir);
+
+    printf("Work directory: %s\n", home);
+    testutil_make_work_dir(home);
+
+    /* Fork a child to create tables and perform operations on them. */
+    memset(&sa, 0, sizeof(sa));
+    sa.sa_handler = sig_handler;
+    testutil_checksys(sigaction(SIGCHLD, &sa, NULL));
+    testutil_checksys((pid = fork()) < 0);
+
+    if (pid == 0) { /* child */
+
+        run_test(home);
+        printf("Child finished processing...\n");
+        return (EXIT_SUCCESS);
+    }
+
+    /* parent */
+    /*
+     * Sleep for the configured amount of time before killing the child. Start the timeout from the
+     * time we notice that child process has written a checkpoint. That allows the test to run
+     * correctly on really slow machines.
+     */
+    while (stat(ckpt_file, &sb) != 0)
+        testutil_sleep_wait(1, pid);
+
+    /* Sleep for a while. Let the child process do some operations on the tables. */
+    sleep(TIMEOUT);
+    sa.sa_handler = SIG_DFL;
+    testutil_checksys(sigaction(SIGCHLD, &sa, NULL));
+
+    /* Kill the child process. */
+    printf("Kill child\n");
+    testutil_checksys(kill(pid, SIGKILL) != 0);
+    testutil_checksys(waitpid(pid, &status, 0) == -1);
+
+    /* Reopen the connection and verify that the tables match each other. */
+    testutil_check(wiredtiger_open(home, &event_handler, conn_config, &conn));
+    testutil_check(conn->open_session(conn, NULL, NULL, &session));
+    verify_tables(session);
+
+    /* Clean-up. */
+    testutil_check(session->close(session, NULL));
+    testutil_check(conn->close(conn, NULL));
+    testutil_cleanup(opts);
+
```

Comment:
`testutil_cleanup` closes `opts->conn` is it's not NULL. Is it initialized in your test? If yes, the pointer is probably invalid after the simulated crash?

Reply:
I never open a connection using `opts->conn` as I have my own local connection context. When the child process crashes, all the memory held by structs is freed implicitly. 

We also never open connections in child and parent in parallel.

Reply:
I see, all good then! :)